### PR TITLE
MINOR: add MockConfigRepository

### PR DESCRIPTION
**Pull request by cmccabe**
Use MockConfigRepository rather than CachedConfigRepository in unit
tests. This is useful for an upcoming change that will remove
CachedConfigRepository.

### CODE REVIEW

File: core/src/test/scala/kafka/server/metadata/MockConfigRepository.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    val properties = configs.getOrDefault(configResource, new Properties())
+    val newProperties = new Properties(properties)
+    newProperties.put(key, value)
+    configs.put(configResource, newProperties)
+  }
+
+  def setTopicConfig(topicName: String, key: String, value: String): Unit = configs.synchronized {
+    setConfig(new ConfigResource(TOPIC, topicName), key, value)
+  }
+}
```
Comment (mumrah) [inline core/src/test/scala/kafka/server/metadata/MockConfigRepository.scala]: nit: missing newline

File: core/src/test/scala/unit/kafka/log/LogManagerTest.scala [hunk 2]
```diff
@@ -564,7 +563,7 @@ class LogManagerTest {
   @Test
   def testBrokerConfigChangeDeliveredToAllLogs(): Unit = {
     logManager.shutdown()
-    val spyConfigRepository = spy(new CachedConfigRepository)
+    val spyConfigRepository = spy(new MockConfigRepository)
```
Comment (mumrah) [inline core/src/test/scala/unit/kafka/log/LogManagerTest.scala:566]: Do we need to "spy" on these instances? Seems like we could put whatever test-only behavior we want into the MockConfigRepository class itself. Or maybe we need to do this for some Mockito reason like "all arguments must be mock/spy-ed"

File: core/src/test/scala/unit/kafka/server/metadata/MockConfigRepositoryTest.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ */
+
+package kafka.server.metadata
+
+import java.util.Properties
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+class MockConfigRepositoryTest {
```
Comment (mumrah) [inline core/src/test/scala/unit/kafka/server/metadata/MockConfigRepositoryTest.scala:25]: So this is a test of the mock class itself?

**Review by mumrah — APPROVED** [review]
LGTM, few minor things inline

File: core/src/test/scala/unit/kafka/log/LogManagerTest.scala (hunk 2 above)
Comment (cmccabe) [reply core/src/test/scala/unit/kafka/log/LogManagerTest.scala:566]: we want to verify that `topicConfig` was called, and this seemed like the easiest way.

File: core/src/test/scala/unit/kafka/server/metadata/MockConfigRepositoryTest.scala (hunk 3 above)
Comment (cmccabe) [reply core/src/test/scala/unit/kafka/server/metadata/MockConfigRepositoryTest.scala:25]: Yeah...

**Pull request merged**